ftw/upgrade/tests/test_migration.py

```diff
@@ -383,7 +384,37 @@ def test_migrate_ownership(self):
 
         self.grant('Manager')
         self.install_profile('plone.app.contenttypes:default')
-        InplaceMigrator('Folder').migrate_object(folder)
+        with self.login(SITE_OWNER_NAME):
+            InplaceMigrator('Folder').migrate_object(folder)
+
+        folder = self.portal.get('the-folder')
+        self.assertTrue(IDexterityContent.providedBy(folder))
+        self.assertEquals('john.doe', folder.Creator())
+        self.assertEquals('peter.pan', folder.getOwner().getId())
+
+    def test_migrate_ownership_no_IOwner(self):
+        john = create(Builder('user').named('John', 'Doe').with_roles('Manager'))
+        peter = create(Builder('user').named('Peter', 'Pan').with_roles('Manager'))
+
+        login(self.portal, john.getId())
+        folder = create(Builder('folder').titled(u'The Folder'))
+        folder.changeOwnership(peter.getUser())
+
+        self.assertTrue(IBaseObject.providedBy(folder))
+        self.assertEquals('john.doe', folder.Creator())
+        self.assertEquals('peter.pan', folder.getOwner().getId())
+
+        self.grant('Manager')
+        self.install_profile('plone.app.contenttypes:default')
+
+        # The creators list behaves differently when the dublin core
+        # behavior is used.
+        self.portal.portal_types['Folder'].behaviors = tuple(
```

Comment:
Do I understand that correctly, that here the behaviour IDublinCore is getting removed from all folder? If thats the case an assertion would maybe help to understand that it is important to have the behaviour removed on folders.

Reply:
yes: the goal of the test is to verify that the creators are migrated properly when the bahavior is not activated on the target type. That's why in the setup of the test we are removing the behavior from the FTI. But it is not the goal that the behavior is removed.
The change is also restored after this test.